js/main.js

```diff
@@ -15,48 +14,98 @@ function getRecipeData(mealType, cusineType) {
   xhrRequest.open('GET', 'https://api.edamam.com/search?app_id=56d29915&app_key=8a2c94a84e7c4a5a2c94ff997508b44b&from=0&to=20&q=&mealType=' + mealType + '&cuisineType=' + cusineType);
   xhrRequest.responseType = 'json';
   xhrRequest.addEventListener('load', function () {
-    console.log('xhrstatus', xhrRequest.status);
-    console.log('xhresponse', xhrRequest.response);
+    var randomIndex = Math.floor(Math.random() * xhrRequest.response.hits.length);
+    var recipeName = xhrRequest.response.hits[randomIndex].recipe.label;
+    var ingredients = xhrRequest.response.hits[randomIndex].recipe.ingredientLines;
+    var imageSrc = xhrRequest.response.hits[randomIndex].recipe.image;
+    var getInstructionsUrl = xhrRequest.response.hits[randomIndex].recipe.url;
+    var recipeObj = {
+      name: recipeName,
+      ingredient: ingredients,
+      imageUrl: imageSrc,
+      instructionsUrl: getInstructionsUrl
+    };
+    recommendedDish(recipeObj);
   });
   xhrRequest.send();
 }
 
+function recommendedDish(recipeObj) {
+  var $dishDescription = document.createElement('div');
+  $dishDescription.setAttribute('class', 'dish-description');
+  $dishDescriptionWrapper.appendChild($dishDescription);
+
+  var $dishImage = document.createElement('img');
+  $dishImage.setAttribute('class', 'dish-image');
+  $dishImage.src = recipeObj.imageUrl;
+  $dishImageWrapper.appendChild($dishImage);
+
+  var $recipeName = document.createElement('h2');
+  $recipeName.textContent = recipeObj.name;
+  $dishDescription.appendChild($recipeName);
+
+  var $ingredientHeading = document.createElement('p');
+  $ingredientHeading.textContent = 'Ingredients';
+  $dishDescription.appendChild($ingredientHeading);
+
+  var $ingredientList = document.createElement('ul');
+  $dishDescription.appendChild($ingredientList);
+
+  for (var k = 0; k < recipeObj.ingredient.length; k++) {
+    var $ingredient = document.createElement('li');
+    $ingredient.textContent = recipeObj.ingredient[k];
+    $ingredientList.appendChild($ingredient);
+  }
+
+  var $getInstructionsLink = document.createElement('a');
+  $getInstructionsLink.textContent = 'Get Instructions';
+  $getInstructionsLink.target = '_blank';
+  $getInstructionsLink.setAttribute('class', 'get-instructions-link');
+  $getInstructionsLink.href = recipeObj.instructionsUrl;
+  $dishDescription.appendChild($getInstructionsLink);
+}
```

Comment:
As much as possible, functions that generate DOM trees should take in the necessary data, and `return` the DOM tree to be used wherever needed, not just append as the last action of the function. I recommend reviewing `dom-creation` to get an idea of what I am referring to. Let me know if you have any questions!